agents/agents-features/agents-features-acp/src/jvmMain/kotlin/ai/koog/agents/features/acp/MessageConverters.kt

```diff
@@ -0,0 +1,205 @@
+                add(
+                    SessionUpdateEvent(
+                        update = SessionUpdate.AgentThoughtChunk(
+                            content = ContentBlock.Text(response.content)
+                        )
+                    )
+                )
+            }
+
+            is Message.Tool.Call -> {
+                add(
+                    SessionUpdateEvent(
+                        update = SessionUpdate.ToolCall(
+                            toolCallId = ToolCallId(response.id ?: UNKNOWN_TOOL_CALL_ID),
+                            // TODO: Support tool description in the event
+                            title = response.tool,
+                            // TODO: Support kind for tools
+                            status = ToolCallStatus.PENDING,
+                            rawInput = response.contentJson,
+                        )
+                    )
+                )
+            }
+        }
+    }
+}
+
+/**
+ * Converts a ContentPart to an ACP ContentBlock.
+ */
+public fun ContentPart.toAcpContentBlock(): ContentBlock {
+    return when (this) {
+        is ContentPart.Text -> {
+            ContentBlock.Text(this.text)
+        }
+
+        is ContentPart.Audio -> {
+            ContentBlock.Audio(
+                data = this.content.toString(),
+                mimeType = this.mimeType,
+            )
+        }
+
+        is ContentPart.File ->
+            when (val content = this.content) {
+                is AttachmentContent.Binary.Base64 -> ContentBlock.Resource(
+                    resource = EmbeddedResourceResource.BlobResourceContents(
+                        blob = content.base64,
+                        // TODO: add uri to the file
+                        uri = UNKNOWN_URI,
+                        mimeType = this.mimeType
+                    )
+                )
+
+                is AttachmentContent.Binary.Bytes -> ContentBlock.Resource(
+                    resource = EmbeddedResourceResource.BlobResourceContents(
+                        blob = content.asBase64(),
+                        // TODO: add uri to the file
+                        uri = UNKNOWN_URI,
+                        mimeType = this.mimeType
+                    )
+                )
+
+                is AttachmentContent.PlainText -> ContentBlock.Resource(
+                    resource = EmbeddedResourceResource.TextResourceContents(
+                        text = content.text,
+                        // TODO: add uri to the file
+                        uri = UNKNOWN_URI
+                    )
+                )
+
+                is AttachmentContent.URL -> {
+                    ContentBlock.ResourceLink(
+                        name = this.fileName ?: UNKNOWN_FILE_NAME,
+                        uri = content.url
+                    )
+                }
+            }
+
+        is ContentPart.Image -> {
+            ContentBlock.Image(
+                data = this.content.toString(),
```

Comment:
Same as with `ContentPart.Audio`. Also, nit, let's move it up next to the `ContentPart.Image` handling, to make it easier to read again, since these are kinda similar and you already put them together in the other conversion extension function.